btf/marshal.go

```diff
@@ -379,19 +384,28 @@ func (e *encoder) deflateVarSecinfos(vars []VarSecinfo) []btfVarSecinfo {
 // The function is intended for the use of the ebpf package and may be removed
 // at any point in time.
 func MarshalMapKV(key, value Type) (_ *Handle, keyID, valueID TypeID, _ error) {
+	if key == nil && value == nil {
```

Comment:
Don't we have `spec.Key != nil || spec.Value != nil` in `createMap`? We shouldn't treat this as a valid input.

Reply:
Sure, this may be overkill, but it's still an exported function. Will remove if you feel strongly about it.